mx.c

```diff
@@ -566,6 +566,7 @@ static int trash_append(struct Mailbox *m)
         // L10N: Displayed if appending to $trash fails when syncing or closing a mailbox
         mutt_error(_("Unable to append to trash folder"));
         m_trash->append = old_append;
```

Comment:
this looks suspicious, why do we set `m_trash->append` before freeing the mailbox?

Reply:
Hi! Thank you for reviewing the PR :). I would like to point out that same happens on line 575-576 on the non error path.

Reply:
We set `Mailbox.append` because we want to change the behaviour for writing,
but we restore it afterwards, because we may not be the only user of _that_ Mailbox.

`Mailbox.append` is an ugly hack that I haven't figured out how to kill yet.

Reply:
> because we may not be the only user of that Mailbox.

oh I see, `mailbox_free()` doesn't unconditionally free the mailbox.